### ISSUE TITLE
Thinkpad L390 - pen & tablet don't appear in Gnome Settings

### ISSUE THREAD

**Issue by sebasmonia** [linuxwacom/libwacom#402]
Hello!
Landed here following a flurry of web searches for these kind of issue. Looking at open & closed bugs I saw #218 for this particular device.
Running Fedora 34 (Gnome 40) I still don't see the tablet & pen config in the Settings. The tip contact is left click, the top button is right click, the middle button has no function. It can be used in apps like Xournal++, though.
Touchscreen also works ok.

Is there anything else to do? Will the lower button ever be configurable in the DE?

Thank you!

**Comment by whot** [comment]
If the tablet is listed in `libwacom-list-local-devices` with the correct properties, its unlikely to be a libwacom issue and more likely to be a gnome issue. What's the output of that tool?

**Comment by sebasmonia** [comment]
The output is:
```
libwacom-list-local-devices
/dev/input/event7 is a tablet but not supported by libwacom
Failed to find any devices known to libwacom.
```

**Comment by whot** [comment]
quick `git grep` doesn't list the L390 but we do support the X390 - have a look at `data/isdv4-51af.tablet` and copy+modify that file accordingly. Or [autogenerate it](https://github.com/linuxwacom/libwacom/wiki/Tablet-Definition-Files#autogenerating-a-tablet-file-with-sysinfosh) which is the best approach since we'll ask you to upload the sysinfo output anyway.

**Comment by sebasmonia** [comment]
Thank you! Following the instructions in the README of the repo, I generated the files.

**Comment by sebasmonia** [comment]
As a test, I tried modifying the existing file by changing the USB id, and I got the tablet & stylus in the settings page.
For the stylus it can only configure one button, the other one is marked as "eraser" and it always produces a right click when pressed while hovering.
The other button never produced a right click despite changing the value in the drop down.

**Comment by whot** [comment]
On most devices, the eraser button is hard-wired to change the tool type to eraser, it doesn't generate a button event at all. There were grand plans to make libinput and xf86-input-wacom hack around this to expose that button as real button but it's too much work and the result is likely to be flaky. So what's likely happening here is that the eraser button cannot be mapped anyway and the other button just has the wrong leader lines in the svg so it points to what is the physical eraser button. Does that sound right?

**Comment by sebasmonia** [comment]
Yes it makes sense!

So in the issue I submitted for the tablet file, do I need to include an SVG for the pen? Are there specific instructions for this?

**Comment by whot** [comment]
hmm, thinking about this again - the stylus issue may not be fixable in libwacom itself. We do have [layout instructions](https://github.com/linuxwacom/libwacom/blob/master/data/layouts/README.md) but those are for the *tablet*, not the stylus. The leader lines in the stylus config panel are part of the control center itself iirc.

If you submit a PR for the new `.tablet` file I'll get that merged asap.

**Comment by sebasmonia** [comment]
My bad, I followed the autogenerate instructions and got confused, submitted the tarball in the other repository instead of a .table here :upside_down_face: 
I guess the stylus button config not responding is, then, a Gnome issue.

Will be away for a few days, but as soon as I'm back I will submit the PR.
Thank you for your help!

**Comment by sebasmonia** [comment]
I'm back home and I see #403 includes the definition for this device.
Guess we can close this once the PR is merged.

Regarding the stylus button, is there is anything to do?

**Comment by whot** [comment]
I merged #403 now, we can sort out the question later, so yay, let's close this one and go to the pub. Or home. Or elsewhere, whatever tickles your fancy.

As for the stylus button - nothing you can do in libwacom itself. You could change the control center so it swaps the SVG leader lines based on `libwacom_stylus_get_eraser_type()`, afaict all styli with `WACOM_ERASER_BUTTON` have the eraser button on the lower one and the secondary button on the higher button (as seen from tip).

**Comment by sebasmonia** [comment]
This one looks correct in Control Center (eraser on top), will check with the Gnome guys first.

Thank you for all your help and the fix.
